src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyResultSet.java

```diff
@@ -510,14 +512,24 @@ private void updateReferenceHaplotype(final Haplotype newHaplotype) {
      */
     public SortedSet<VariantContext> getVariationEvents(final int maxMnpDistance) {
         ParamUtils.isPositiveOrZero(maxMnpDistance, "maxMnpDistance may not be negative.");
-        if (variationEvents == null) {
-            final List<Haplotype> haplotypeList = getHaplotypeList();
-            EventMap.buildEventMapsForHaplotypes(haplotypeList, fullReferenceWithPadding, paddedReferenceLoc, debug, maxMnpDistance);
-            variationEvents = EventMap.getAllVariantContexts(haplotypeList);
+
+        final boolean sameMnpDistance = lastMaxMnpDistanceUsed.isPresent() && maxMnpDistance == lastMaxMnpDistanceUsed.getAsInt();
```

Comment:
How would the MNP distance change within the same tool execution?

Reply:
It shouldn't ever change, but I lean toward keeping the logic self-consistent without assuming anything about the tools that invoke it.  Of course, some of these classes are so entwined with HC and M2 that it's safe to assume things, but in this case the price of caution is small.

Also, the code for exactly when event maps are cached was quite brittle in that you could write reasonable code and encounter gotchas, so I felt like making it all more explicit as to what had and had not been computed previously.